src/wallet/wallet.cpp

```diff
@@ -3633,7 +3633,7 @@ ScriptPubKeyMan* CWallet::AddWalletDescriptor(WalletDescriptor& desc, const Flat
         }
 
         CTxDestination dest;
-        if (!internal && ExtractDestination(script_pub_keys.at(0), dest)) {
+        if (!internal && ExtractDestination(*script_pub_keys.begin(), dest)) {
```

Comment:
220fe61f70f6093a695fd8b6e38e928b63c2b6ad: shouldn't this be a loop over all elements anyway, for `combo()`. It seems it would always label the `P2PK` variant before (which `ExtractDestination` turns into `P2PKH`), and a random one after.

Reply:
That broke `tool_wallet.py —descriptors` because `Address Book: ` needs to be 3 now instead of 1.

Reply:
Extracted to a separate commit and fixed the test.